src/commands.ts

```diff
@@ -981,10 +982,79 @@ async function configCMD(shell: Shell) {
     await shell.openConfig()
 }
 async function supportCMD(shell: Shell) {
-    shell.t.write("Sorry things are not working well. Please ")
-    shell.t.writeln("\x1B]8;;https://github.com/tuzig/terminal7/issues/new?template=bug_report.md\x07report a bug\x1B]8;;\x07")
-    shell.t.writeln("or talk to us on our \x1B]8;;https://discord.gg/Puu2afdUtr\x07discord server\x1B]8;;\x07")
-}
+    shell.t.write("Apologies. Have you tried resetting the App?\n")
+    shell.t.writeln("If that doesn't work, please send us a log of the error.")
+    const insecure = terminal7.conf.peerbook.insecure,
+    schema = insecure?"http":"https"
+    
+    // ask for email address + check validity
+    let email = terminal7.conf.peerbook.email || (await shell.askValue("Enter your email address"))
+    while (!email.match(/^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$/)) {
+        shell.t.writeln("Invalid email address")
+        email = await shell.askValue("Enter your email address")
+    }
+
+    // Menu to ask user if they want to send the log or Post it to mail
+    const fieldsNative = [
+        {prompt: "Copy log to clipboard"},
+        {prompt: "Send log to support"},
+        {prompt: "Save log to file"},
+        {prompt: "Cancel"}
+    ]
+
+    // Ask user for choice
+    !Capacitor.isNativePlatform() ? fieldsNative.splice(2,1) : fieldsNative
+    const choice = await shell.runForm(fieldsNative,"menu" , "Choose an option")
+   // Switch case to handle user choice
+    switch (choice) {
+        case "Copy log to clipboard":
+            // Saving the log to the clipboard
+            const log = terminal7.dumpLog()
+            await Clipboard.write({string: (await log).toString()})
+            terminal7.notify("Log copied to clipboard.")
+            shell.t.writeln("Please paste into discord support channel.")
+            break
+        case "Send log to support":
+            const description = await shell.askValue("please describe the issue")
+            shell.t.writeln("Sending log to support...")
+            const res = await fetch(`${schema}://${terminal7.conf.net.peerbook}/support`, {
+                method: "POST",
+                body: JSON.stringify({
+                    email,
+                    log: (await Clipboard.read()).value,
```

Comment:
Something is weird here. I remember you fixed it so it won't go through the clipboard and destroy it's old value.

Reply:
Yeah i forgot to change the line to await terminal7.dumpLog()...